packages/ui/src/components/Visualization/Canvas/FormV2/fields/BeanField.tsx

```diff
@@ -0,0 +1,148 @@
+import { FunctionComponent, useCallback, useContext, useMemo, useState } from 'react';
+import { BeansEntityHandler } from '../../../../../models/visualization/metadata/beans-entity-handler';
+import { EntitiesContext } from '../../../../../providers';
+import { getSerializedModel } from '../../../../../utils';
+import { NewBeanModal } from '../../../../Form';
+import { Typeahead } from '../../../../typeahead/Typeahead';
+import { TypeaheadItem } from '../../../../typeahead/Typeahead.types';
+import { useFieldValue } from '../hooks/field-value';
+import { SchemaContext } from '../providers/SchemaProvider';
+import { FieldProps } from '../typings';
+import { extractGroup } from '../../../../../utils/get-tagged-field-from-string';
+
+export const BeanField: FunctionComponent<FieldProps> = ({ propName, required }) => {
+  const { schema } = useContext(SchemaContext);
+  const { value = '', onChange } = useFieldValue<string>(propName);
+  const entitiesContext = useContext(EntitiesContext);
+  const camelResource = entitiesContext?.camelResource;
+  const beanReference = value;
+  const beansHandler = useMemo(() => {
+    return new BeansEntityHandler(camelResource);
+  }, [camelResource]);
+  const beanSchema = useMemo(() => {
+    return beansHandler.getBeanSchema();
+  }, [beansHandler]);
+  const [isNewBeanModalOpen, setIsNewBeanModalOpen] = useState<boolean>(false);
+  const [inputValue, setInputValue] = useState<string>(beanReference);
+  const beansEntity = useMemo(() => {
+    return beansHandler.getBeansEntity();
+  }, [beansHandler]);
+
+  const items: TypeaheadItem<string | undefined>[] =
+    beansEntity?.parent.beans.map((item) => ({
+      name: item.name ? (beansHandler.getReferenceFromName(item.name) ?? '') : '',
+      description: String(item.type),
+      value: String(item.name),
+    })) ?? [];
+
+  const selectedItem = useMemo(() => {
+    if (!value) {
+      return undefined;
+    }
+    // Object values are stringified. Double check later different approach.
+    if (typeof value === 'object') {
+      return {
+        value: JSON.stringify(value),
+        name: JSON.stringify(value),
+        description: '',
+      };
+    }
+    return items.find((item) => item.name === value) ?? { value: value, name: value, description: '' };
+  }, [value]);
+
+  const onItemChange = useCallback(
+    (item?: TypeaheadItem<string>) => {
+      onChange(item!.name);
+    },
+    [onChange],
+  );
+
+  const onCleanInput = useCallback(() => {
+    onChange('');
+  }, [onChange]);
+
+  const createNewWithNameValue = 'create-new-bean-with-name';
+  const id = `${propName}-popover`;
+
+  const onSelect = useCallback(
+    (value: string | undefined, filterValue: string | undefined) => {
+      if (value) {
+        if (value === createNewWithNameValue) {
+          setInputValue(filterValue ?? '');
+          setIsNewBeanModalOpen(true);
+        } else {
+          setInputValue('');
+          setIsNewBeanModalOpen(true);
+        }
```

Comment:
What do you think about something like this?
```suggestion
        if (value === createNewWithNameValue) {
          setInputValue(filterValue ?? '');
        } else {
          setInputValue('');
        }

        setIsNewBeanModalOpen(true);
```